src/app/themes.tsx

```diff
@@ -10,51 +10,113 @@ import {
     createTheme as createMaterialTheme,
 } from "@mui/material";
 
-type CustomTheme = {
-    foregroundColor: string;
-    backgroundColor: string;
-    primaryForegroundColor: string;
-    primaryBackgroundColor: string;
-    secondaryForegroundColor: string;
-    secondaryBackgroundColor: string;
-    accentBackgroundColor: string;
-    accentForegroundColor: string;
-    surfaceForegroundColor: string;
-    surfaceBackgroundColor: string;
-    errorColor: string;
+export type standardPalette = {
+    background: string[];
+    foreground: string[];
+    largeForeground: string[];
 };
 
-const lightTheme: CustomTheme = {
-    foregroundColor: "#000000",
-    backgroundColor: "#eeeeee",
-    primaryForegroundColor: "#000000",
-    primaryBackgroundColor: "#63a036",
-    secondaryForegroundColor: "#ffffff",
-    secondaryBackgroundColor: "#066940",
-    accentBackgroundColor: "#609fd3",
-    accentForegroundColor: "#000000",
-    surfaceForegroundColor: "#000000",
-    surfaceBackgroundColor: "#ffffff",
-    errorColor: "#ff624e",
+export type rainbowPalette = {
+    red: standardPalette;
+    orange: standardPalette;
+    yellow: standardPalette;
+    green: standardPalette;
+    blue: standardPalette;
+    purple: standardPalette;
+    grey: standardPalette;
+    brown: standardPalette;
 };
 
-const darkTheme: CustomTheme = {
-    foregroundColor: "#ffffff",
-    backgroundColor: "#1a1a1a",
-    primaryForegroundColor: "#ffffff",
-    primaryBackgroundColor: "#066940",
-    secondaryForegroundColor: "#000000",
-    secondaryBackgroundColor: "#63a036",
-    accentBackgroundColor: "#1b629c",
-    accentForegroundColor: "#ffffff",
-    surfaceForegroundColor: "#ffffff",
-    surfaceBackgroundColor: "#161414",
-    errorColor: "#ff624e",
+export type customTheme = {
+    light: boolean;
+    main: standardPalette & { lighterForeground: string; border: string };
+    primary: standardPalette;
+    error: string;
+    shadow: string;
+    rainbow: rainbowPalette;
+};
+
+export const lightTheme: customTheme = {
```

Comment:
I think it would be more readable to put it in array, as the future user can simply look at the element position in the array and map it (I could use stuff like `new Array(2).fill("#000000")`, but
1. There are only two elements
2. Less readable
3. Not all are filled with #000000, so eventually it makes things more complicated

Reply:
Yeah I agree the `new Array(2).fill("#000000")` is less readable. 

I think I quite like paopao's idea though.

Reply:
Nevermind -> talked in person and I think with how it's refactored (with the `BLACK` and `WHITE` set as constants, it doesn't need the default settings.